src/OpenTelemetry.Api/Baggage.cs

```diff
@@ -345,17 +345,17 @@ public override int GetHashCode()
         {
             var baggage = this.baggage ?? EmptyBaggage;
 
-            unchecked
+            var hash = 17;
+            foreach (var item in baggage)
```

Comment:
Could be another PR, but in a few places we compute the hash in the constructor since often the state is immutable. I think Baggage is one such case we could do this.

Reply:
We might want to be selective about where we do that. Because, GetHashCode is probably never called on Baggage. Unless someone was using it as a key somewhere, which seems unlikely? We really only want to do that computation ahead of time where we know for sure it will be used as key. Otherwise it would be just wasted cycles 😄

Reply:
Good point, makes sense.